compiler/AST/TryStmt.cpp

```diff
@@ -52,30 +70,61 @@ void TryStmt::accept(AstVisitor* visitor) {
     if (_body) {
       _body->accept(visitor);
     }
+
+    for_alist(c, _catches) {
+      CatchStmt* catchStmt = toCatchStmt(c);
+      catchStmt->accept(visitor);
+    }
+
     visitor->exitTryStmt(this);
   }
 }
 
-Expr* TryStmt::copy(SymbolMap* map, bool internal) {
-  return NULL;
-}
-
-Expr* TryStmt::copyInner(SymbolMap* map) {
-  return NULL;
+TryStmt* TryStmt::copyInner(SymbolMap* map) {
+  TryStmt* copy = new TryStmt(_tryBang, COPY_INT(_body), NULL);
+  for_alist(c, _catches) {
+    copy->_catches.insertAtTail(c->copy());
+  }
+  return copy;
 }
 
 void TryStmt::replaceChild(Expr* old_ast, Expr* new_ast) {
-
+  if (old_ast == _body) {
+    _body = toBlockStmt(new_ast);
+  } else if (old_ast->list == &_catches) {
+    old_ast->replace(new_ast);
+  }
 }
 
 Expr* TryStmt::getFirstChild() {
   return _body;
 }
 
 Expr* TryStmt::getFirstExpr() {
-  return _body->getFirstExpr();
+  if (_body) {
+    return _body->getFirstExpr();
+  }
+  return NULL;
 }
 
 Expr* TryStmt::getNextExpr(Expr* expr) {
-  return this;
+  Expr* retVal = this;
+
+  if (expr == _body) {
+    if (Expr* firstCatch = _catches.first()) {
+      retVal = firstCatch->getFirstExpr();
+    }
+  } else if (expr->list == &_catches) {
+    if (Expr* nextCatch = expr->next) {
+      retVal = nextCatch->getFirstExpr();
+    }
+  }
+
+  return retVal;
+}
+
+GenRet TryStmt::codegen() {
+  codegenStmt(this);
+
+  return _body->codegen();
 }
```

Comment:
I have no idea what GenRet(-1) is. I'd normally just return an empty one, like this
GenRet ret;
return ret;